src/Appwrite/Platform/Tasks/Backup.php

```diff
@@ -0,0 +1,318 @@
+
+        $backups = $local->getRoot() . '/files';
+        $tarFile = $local->getPath($filename);
+
+        $this->fullBackup($backups);
+        $this->tar($backups, $tarFile);
+        $this->upload($tarFile, $local);
+
+        self::log('--- Backup Finish ' . (microtime(true) - $start) . ' seconds --- '   . PHP_EOL . PHP_EOL);
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function fullBackup(string $target)
+    {
+        $start = microtime(true);
+        self::log('Xtrabackup start');
+
+        if (!file_exists(self::BACKUPS_PATH)) {
+            throw new Exception('Mount directory does not exist');
+        }
+
+        if (!file_exists($target) && !mkdir($target, 0755, true)) {
+            throw new Exception('Error creating directory: ' . $target);
+        }
+
+        $filename = basename($target);
+        $logfile = $target . '/../backup.log';
+
+        $args = [
+            'xtrabackup',
+            '--user=' . $this->dsn->getUser(),
+            '--password=' . $this->dsn->getPassword(),
+            '--host=' . $this->dsn->getHost(),
+            '--port=' . $this->dsn->getPort(),
+            '--backup',
+            '--strict',
+            '--history="' . $this->database . '|' . pathinfo($filename, PATHINFO_FILENAME) . '"', // PERCONA_SCHEMA.xtrabackup_history
+            '--slave-info',
+            '--safe-slave-backup',
+            '--safe-slave-backup-timeout=300',
+            '--check-privileges', // checks if Percona XtraBackup has all the required privileges.
+            '--target-dir=' . $target,
+            '--compress=' . self::COMPRESS_ALGORITHM,
+            '--compress-threads=' . $this->processors, // Better not using all processors
+            '--parallel=' . $this->processors,
+            '--rsync', // https://docs.percona.com/percona-xtrabackup/8.0/accelerate-backup-process.html
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec ' . $this->xtrabackupContainerId . ' ' . implode(' ', $args);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+        if (!str_contains($stderr, 'completed OK!')) {
+            shell_exec('rm -rf ' . $target);
+            throw new Exception(' Backup failed: ' . $stderr);
+        }
+
+        if (!unlink($logfile)) {
+            throw new Exception('Error deleting: ' . $logfile);
+        }
+
+        self::log('Xtrabackup end ' . (microtime(true) - $start) . ' seconds');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function tar(string $directory, string $file)
+    {
+        $start = microtime(true);
+        self::log('Tar start');
+
+        $stdout = '';
+        $stderr = '';
+        $cmd = 'cd ' . $directory . ' && tar zcf ' . $file . ' . && cd ' . getcwd();
+        Console::execute($cmd, '', $stdout, $stderr);
+        if (!empty($stderr)) {
+            throw new Exception('Tar failed:' . $stderr);
```

Comment:
Let's add a retry loop here with 5 max attempts before throwing

Reply:
Done! 
We will have to adjust the attempts